graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/node/GqlJcrProperty.java

```diff
@@ -341,7 +341,7 @@ public Double getFloatValue() {
     @GraphQLDescription("GraphQL representation of the node this property references in case the property is single-valued, null otherwise")
     public GqlJcrNode getRefNode() throws GqlJcrUnresolvedNodeReferenceException {
         try {
-            if (property.isMultiple()) {
+            if (property.isMultiple() || property.getType() != PropertyType.REFERENCE || property.getType() != PropertyType.WEAKREFERENCE) {
```

Comment:
In which case could `(property.getType() != PropertyType.REFERENCE || property.getType() != PropertyType.WEAKREFERENCE)` ever be false ?

Reply:
no idea, it's what was suggested in the ticket.

Reply:
And when I tested in the node properties:
- the j:nodename for example it's of type reference

Reply:
Hint : it's not possible to have property.getType() (a number) equals to PropertyType.REFERENCE and PropertyType.WEAKREFERENCE at the same time. So the opposite condition will always be true, hence the condition is broken (can be simplified to if (true) ) and you will never get a value anymore when calling getRefNode